File: app/src/main/java/dev/bee/kanjianki/MainActivitySettingsAnkiSource.java
package dev.bee.kanjianki;

import android.view.View;
import android.widget.LinearLayout;

import dev.bee.kanjianki.core.RecordsSyncModels;

final class MainActivitySettingsAnkiSource {
    private final MainActivitySettingsAnkiSourceInputs inputs;
    private final MainActivitySettingsAnkiSourceFrequencyRange frequencyRange;
    private final MainActivitySettingsAnkiSourceNoteType noteType;
    private final MainActivitySettingsAnkiSourceImportFilters importFilters;

    MainActivitySettingsAnkiSource(MainActivitySettings activity) {
        this.inputs = new MainActivitySettingsAnkiSourceInputs(activity);
        MainActivitySettingsAnkiSourceValidation validation = new MainActivitySettingsAnkiSourceValidation(activity);
        this.frequencyRange = new MainActivitySettingsAnkiSourceFrequencyRange(activity, this.inputs, validation);
        this.noteType = new MainActivitySettingsAnkiSourceNoteType(activity);
        this.importFilters = new MainActivitySettingsAnkiSourceImportFilters(activity, this.inputs, validation);
    }

    View noteTypeSettingsPanel(RecordsSyncModels.Settings current) {
        return noteType.noteTypeSettingsPanel(current);
    }

    View importFilterSettingsPanel(RecordsSyncModels.Settings current) {
        return importFilters.importFilterSettingsPanel(current);
    }

    View frequencyRangeSettingsPanel(RecordsSyncModels.Settings current) {
        return frequencyRange.frequencyRangeSettingsPanel(current);
    }
}

File: app/src/main/kotlin/dev/bee/kanjianki/MainActivitySettingsAnkiSource.kt
package dev.bee.kanjianki

import dev.bee.kanjianki.core.RecordsSyncModels

internal class MainActivitySettingsAnkiSource(activity: MainActivitySettings) {
    private val validation = MainActivitySettingsAnkiSourceValidation(activity)
    private val frequencyRange = MainActivitySettingsAnkiSourceFrequencyRange(activity, validation)
    private val noteType = MainActivitySettingsAnkiSourceNoteType(activity)
    private val importFilters = MainActivitySettingsAnkiSourceImportFilters(activity, validation)

    fun noteTypeSettingsPanelModel(current: RecordsSyncModels.Settings): SettingsNoteTypePanelModel {
        return noteType.noteTypeSettingsPanelModel(current)
    }

    fun importFilterSettingsPanelModel(
        current: RecordsSyncModels.Settings,
        tagRepairedCards: Boolean,
    ): SettingsImportFiltersPanelModel {
        return importFilters.importFilterSettingsPanelModel(current, tagRepairedCards)
    }

    fun frequencyRangeSettingsPanelModel(current: RecordsSyncModels.Settings): SettingsFrequencyRangePanelModel {
        return frequencyRange.frequencyRangeSettingsPanelModel(current)
    }
}
